tests/unit/factories.py

```diff
@@ -46,3 +47,60 @@ class Meta:
     port: int = factory.LazyAttribute(lambda n: random.randrange(1024, 65536))
     rsa_fingerprint: str = factory.fuzzy.FuzzyText(prefix="SHA256:")
     ed25519_fingerprint: str = factory.fuzzy.FuzzyText(prefix="SHA256:")
+
+
+class MockGithubRunnerCharmUnitFactory(factory.Factory):
+    class Meta:
+        model = MagicMock
```

Comment:
What is the reason for using `MagicMock` here please? Usually you would set it as the class being created, you can get the types to work as demonstrated on discourse-gatekeeper: https://github.com/canonical/discourse-gatekeeper/blob/5c6f9bedadf2137b9e6f93ed2b50fcdb58fca084/tests/factories.py#L22